### PR TITLE
refactor: centralizes returnTo stack navigation

### PR DESCRIPTION
**Pull request by ygrishajev**
Introduces UrlReturnToStack + useReturnTo to preserve multi-step returnTo across onboarding/login/Auth0 (SSR + API routes), with legacy `from` fallback.

## Summary by CodeRabbit

* **New Features**
  * Multi-entry return-to navigation stack and hook for consistent back/return behavior across auth, onboarding, and deployment.

* **Refactor**
  * Unified navigation flow: sign-in/sign-up, onboarding, and deployment now use the same return/navigation API; some deployment-specific UI paths streamlined.

* **Bug Fixes**
  * More reliable redirects and edge-case handling for external auth and repo-driven flows.

* **Tests**
  * New/updated tests covering the return-to stack and navigation behaviors.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/deploy-web/src/hooks/useReturnTo/useReturnTo.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    const { result } = setup({
+      windowAvailable: false,
+      stackReturnTo: "/ignored"
+    });
+
+    expect(() => {
+      result.current.navigateBack();
+    }).toThrow("No returnTo found");
+  });
+
```
Comment (baktun14) [inline apps/deploy-web/src/hooks/useReturnTo/useReturnTo.spec.tsx:81]: question: does it check for absolute paths?

```suggestion
+  it("sanitizes absolute returnTo URLs to relative paths", () => {
+    const { result, routerPush } = setup({
+      pathname: "/current",
+      search: "q=1",
+      stackReturnTo: "https://evil.com/attack?x=1"
+    });
+
+    // hook should strip scheme/host and only expose pathname + search
+    expect(result.current.returnTo).toBe("/attack?x=1");
+
+    act(() => {
+      result.current.navigateBack();
+    });
+
+    expect(routerPush).toHaveBeenCalledWith("/attack?x=1");
+  });
```

**Review by baktun14 — APPROVED** [review]

File: apps/deploy-web/src/hooks/useReturnTo/useReturnTo.spec.tsx (hunk 1 above)
Comment (ygrishajev) [reply apps/deploy-web/src/hooks/useReturnTo/useReturnTo.spec.tsx:81]: yes, [here](https://github.com/akash-network/console/blob/079f0d4640aa3936d6ef337e46ae4bcb2a8104d4/apps/deploy-web/src/utils/getValidInternalReturnToUrl/getValidInternalReturnToUrl.ts#L10)

Comment (baktun14) [reply apps/deploy-web/src/hooks/useReturnTo/useReturnTo.spec.tsx:81]: Perfect
